erdjs-snippets/lending-pool/lendingPoolInteractor.ts

```diff
@@ -0,0 +1,348 @@
+    }
+
+    async deployDummyLiquidityPool(deployer: ITestUser, tokenIdentifier: string): Promise<{ address: IAddress, returnCode: ReturnCode }> {
+        // Load the bytecode from a file.
+        let code = await loadCode(PathToWasmDummyLiquidityPool);
+
+        // Prepare the deploy transaction.
+        let transaction = this.contract.deploy({
+            code: code,
+            codeMetadata: new CodeMetadata(),
+            initArguments: [
+                new TokenIdentifierValue(tokenIdentifier),
+                new BigIntValue(0),
+                new BigIntValue(40000000),
+                new BigIntValue(1000000000),
+                new BigIntValue(800000000),
+                new BigIntValue(100000000),
+                new BigIntValue(700000000)
+            ],
+            gasLimit: 100000000,
+            chainID: this.networkConfig.ChainID
+        });
+
+        // Set the transaction nonce. The account nonce must be synchronized beforehand.
+        // Also, locally increment the nonce of the deployer (optional).
+        transaction.setNonce(deployer.account.getNonceThenIncrement());
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await deployer.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
+        let address = SmartContract.computeAddress(transaction.getSender(), transaction.getNonce());
+
+        // Let's broadcast the transaction and await its completion:
+        await this.networkProvider.sendTransaction(transaction);
+        let transactionOnNetwork = await this.transactionWatcher.awaitCompleted(transaction);
+
+        // In the end, parse the results:
+        let { returnCode } = this.resultsParser.parseUntypedOutcome(transactionOnNetwork);
+
+        console.log(`LendingPoolInteractor.deployDummyLiquidityPool(): contract = ${address}`);
+        return { address, returnCode };
+    }
+
+    async deploy(deployer: ITestUser, dummyAddress: IAddress): Promise<{ address: IAddress, returnCode: ReturnCode }> {
+        // Load the bytecode from a file.
+        let code = await loadCode(PathToWasm);
+
+        // Prepare the deploy transaction.
+        let transaction = this.contract.deploy({
+            code: code,
+            codeMetadata: new CodeMetadata(),
+            initArguments: [
+                new AddressValue(dummyAddress)
+            ],
+            gasLimit: 100000000,
+            chainID: this.networkConfig.ChainID
+        });
+
+        // Set the transaction nonce. The account nonce must be synchronized beforehand.
+        // Also, locally increment the nonce of the deployer (optional).
+        transaction.setNonce(deployer.account.getNonceThenIncrement());
+
+        // Let's sign the transaction. For dApps, use a wallet provider instead.
+        await deployer.signer.sign(transaction);
+
+        // The contract address is deterministically computable:
```

Comment:
This was the place where `contract.setAddress()` would make sense. You can remove `setAddress()` from functions below. And perhaps add it once, only here.